app/services/search/postgres/reading_list.rb

```diff
@@ -3,17 +3,23 @@ module Postgres
     class ReadingList
       DEFAULT_STATUSES = %w[valid confirmed].freeze
 
-      def self.search_documents(user, term: nil, status: nil, page: 1, per_page: 60)
+      def self.search_documents(user, term: nil, status: nil, tags: nil, tags_mode: :any, page: 1, per_page: 60)
         status ||= DEFAULT_STATUSES
         page = (page || 1).to_i
         per_page = [(per_page || 60).to_i, 100].min
+        tagged_with_any = tags_mode.to_sym == :any
 
         reading_list_article_ids = user.reactions.readinglist.where(status: status)
           .order(created_at: :desc)
           .pluck(:reactable_id, :id)
 
         article_ids = reading_list_article_ids.map { |article_id, _reaction_id| article_id }
+
         articles = if term.present?
+                     # NOTE: [@rhymes] I'm not sure how to combine `pg_search`'s multisearch
```

Comment:
Unfortunately, I'm not completely sure either, as I've never used multi-search before, only `pg_search_scope` 😒 Maybe with a manual join pm` searchable_type` and `searchable_id`?

I actually did some digging now, [this issue](https://github.com/Casecommons/pg_search/issues/33) has some pointers including this interesting tidbit:

> It's not well documented, but `PgSearch.multisearch` really just calls through to a `pg_search_scope` called `PgSearch::Document.search`

Reply:
Interesting! I'm sure we can whip up something with a left join. Need to dig a bit into this